numpy/core/tests/test_indexerrors.py

```diff
@@ -110,6 +111,13 @@ def assign(obj, ind, val):
         assert_raises(IndexError, lambda: a[(1, [0, 1])])
         assert_raises(IndexError, lambda: assign(a, (1, [0, 1]), 1))
 
+    def test_mapping_error_message(self):
+        a = np.zeros((3, 5))
+        index =(2, 4, 0)
+        with pytest.raises(IndexError,
+                match=f"array has {a.ndim}-dimension, but {len(index)} were indexed"):
```

Comment:
same here: for styling, your code editor likely has some options to automatically check for PEP8 in real time. If you don't know, let me know, and I can help you setting it up.